client.go

```diff
@@ -0,0 +1,200 @@
+		responseMode:    ResponseModeEverything,
+		fixLogFactory:   quickfix.NewNullLogFactory(),
+	}
+}
+
+type NewClientOption func(o *Options)
+
+func WithMessageHandlingOpt(mh MessageHandling) NewClientOption {
+	return func(o *Options) {
+		o.messageHandling = mh
+	}
+}
+
+func WithResponseModeOpt(rm ResponseMode) NewClientOption {
+	return func(o *Options) {
+		o.responseMode = rm
+	}
+}
+
+func WithZapLogFactory(logger *zap.SugaredLogger) NewClientOption {
+	return func(o *Options) {
+		o.fixLogFactory = NewZapLogFactory(logger)
+	}
+}
+
+type Client struct {
+	l           *zap.SugaredLogger
+	mu          sync.Mutex
+	isConnected atomic.Bool
+	initiator   *quickfix.Initiator
+	pending     map[string]*call
+
+	apiKey       string
+	privateKey   ed25519.PrivateKey
+	beginString  string
+	targetCompID string
+	senderCompID string
+
+	options Options
+}
+
+func NewClient(l *zap.SugaredLogger, conf Config, opts ...NewClientOption) (*Client, error) {
+	// Get BeginString, TargetCompID and SenderCompID from settings.
+	if conf.Settings == nil {
+		return nil, errors.New("empty quickfix settings")
+	}
+
+	globalSettings := conf.Settings.GlobalSettings()
+	beginString, err := globalSettings.Setting("BeginString")
+	if err != nil {
+		l.Errorw("Failed to read BeginString from settings", "error", err)
+		return nil, err
+	}
+	targetCompID, err := globalSettings.Setting("TargetCompID")
+	if err != nil {
+		l.Errorw("Failed to read TargetCompID from settings", "error", err)
+		return nil, err
+	}
+	senderCompID, err := globalSettings.Setting("SenderCompID")
+	if err != nil {
+		l.Errorw("Failed to read SenderCompID from settings", "error", err)
+		return nil, err
+	}
+
+	privateKey, err := GetEd25519PrivateKeyFromFile(conf.PrivateKeyFilePath)
+	if err != nil {
+		l.Errorw("Failed to GetEd25519PrivateKeyFromFile", "error", err)
+		return nil, err
+	}
+
+	options := defaultOpts()
+	for _, opt := range opts {
+		opt(&options)
+	}
+
+	// Create a new Client object.
+	client := &Client{
+		l:            l,
+		pending:      make(map[string]*call),
+		apiKey:       conf.APIKey,
+		privateKey:   privateKey,
+		beginString:  beginString,
+		targetCompID: targetCompID,
+		senderCompID: senderCompID,
+		options:      options,
+	}
+
+	// Init session and logon to Binance FIX API server.
+	client.initiator, err = quickfix.NewInitiator(
+		client,
+		quickfix.NewMemoryStoreFactory(),
+		conf.Settings,
+		options.fixLogFactory,
+	)
+	if err != nil {
+		client.l.Errorw("Failed to create new initiator", "error", err)
+		return nil, err
+	}
+
+	err = client.Start()
+	if err != nil {
+		client.l.Errorw("Failed to start fix connection", "error", err)
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func (c *Client) Start() error {
+	if err := c.initiator.Start(); err != nil {
+		c.l.Errorw("Failed to initialize initiator", "error", err)
+		return err
+	}
+
+	// Wait for the session to be authorized by the server.
+	for !c.IsConnected() {
```

Comment:
Should add timeout for it in case there is something bad happens.

Reply:
Yeah, I updated at [d7c284d](https://github.com/KyberNetwork/binance_fix_api/pull/1/commits/d7c284d8b95029ff24dd7a3ffdaf51b761f8926a)